### ISSUE TITLE
Name matching service returns results from old dataset versions

### ISSUE THREAD

**Issue by miahi** [CatalogueOfLife/backend#1103]
Since the CoL maintenance around 20th of Jan I see some inconsistent behaviour in the name matching.

Basically what happens is that I get old IDs (from dataset 3) for some of the species whilst matching as the first result, even though the "dataset" ID is set in the request.

https://api.catalogueoflife.org/nidx/match?q=Agrostis%20tileni
...
id	817314
...

https://api.catalogueoflife.org/dataset/2368/nameusage?nidx=817314
...
datasetKey        3
id           "15e466ac-6606-46e4-b052-70ee8304fbda"
sectorKey           85
scientificName  "Agrostis tileni"
...

This also seems to take the name-match (https://data.catalogueoflife.org/tools/name-match) UI by surprise, it will not display the result correctly (the requests above are what the UI is sending to the server). 

To reproduce, upload this CSV to the https://data.catalogueoflife.org/tools/name-match tool with default settings: 

scientificName
Agrostis tileni
Lotus glaucus

The first species does not have the taxonomic data shown; the second one is fine.

The same test worked before the upgrade.

Thank you,
Mihai

**Comment by mdoering** [comment]
The lookup is also really slow, I'll have to re-optimize the index use

**Comment by mdoering** [comment]
@miahi will be a few hours until the fix is deployed

**Comment by mdoering** [comment]
sorry, took a while. Its live now

**Comment by miahi** [comment]
Thank you! I agree it's faster, but now I get no match at all for Agrostis tileni. 
https://api.catalogueoflife.org/nidx/match?q=Agrostis%20tileni
{"type":"none"}

Is this maybe related to a new version of CoL checklist? It seems to exist on the main site as 
https://www.catalogueoflife.org/data/taxon/65THT

**Comment by mdoering** [comment]
Ouch. I improved the gender stemming of latin epithets which the matching is based on. That now includes stemming for `-i` and `-is`. For that to work I need to rebuild the names index though, I did not think of that when deploying as its the first time this happens. I am afraid this will take me some days to fix. For names without stemming, e.g. monomials or the classic latin -a/-us/-um endings the matching should works as it did. Sorry for the bad fix, I did not forsee the consequences of that change.
